### PR TITLE
mtx. update deviceType processing

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: tests/mtx/device.type.test.ts [hunk 1]
```diff
@@ -116,8 +440,9 @@ describe(`MTX device types (${deviceTypes.length})`, () => {
             let buffer = new CommandBinaryBuffer(hex);
             const deviceType = buffer.getDeviceType();
 
-            expect(deviceType.type).toBe(type);
-            expect(deviceType.revision).toBe(revision);
+            expect(deviceType).toBeDefined();
+            expect(deviceType?.type).toBe(type);
```
Comment (DarkPark) [inline tests/mtx/device.type.test.ts:444]: seems `deviceType.type` should be always present

File: src/mtx/utils/deviceType.ts [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+const nibbles5 = ['.', 'A', 'B', 'C', 'D', 'E', 'F', 'H', 'K', 'G'];
+const nibbles6 = ['.', '1', '2', '3', '4'];
+const nibbles7 = ['.', 'L', 'M', 'Z', 'K'];
+const nibbles8 = ['.', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+const nibbles9 = ['.', 'D', 'B', 'C', 'E', 'P', 'R', 'O', 'L', 'F', 'S', 'M', 'Y', 'G', 'N', 'U'];
+const nibbles10 = ['.', '0', '1', '2', '3', '4', '5', '6', 'P', 'R', 'L', 'E', 'G', '-', '/'];
+const nibbles11 = ['.', 'H', 'A', 'T', '0', '0', '0', '0', '0', '1', '2', '3', '4', '0', '0', '0'];
+const nibbles12 = ['.', '0', '1', '2', '3', '4', '5', '6', '7', '8', 'I', 'X', 'G', 'W', 'M', '-'];
+
+const splitDeviceTypeByte = ( byte: number ): Array<number> => [
```
Comment (DarkPark) [inline src/mtx/utils/deviceType.ts:19]:
```suggestion
const splitByte = ( byte: number ): Array<number> => [
```

? as it's not very DeviceType-specific

File: src/mtx/utils/deviceType.ts [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    byte >> 4,
+    byte & 0x0F
+];
+
+export interface IDeviceType {
```
Comment (DarkPark) [inline src/mtx/utils/deviceType.ts:24]: would be nice to place it before consts

File: tests/mtx/device.type.test.ts (hunk 1 above)
Comment (Pnom) [reply tests/mtx/device.type.test.ts]: fixed

File: src/mtx/utils/deviceType.ts (hunk 2 above)
Comment (Pnom) [reply src/mtx/utils/deviceType.ts]: fixed

File: src/mtx/utils/deviceType.ts (hunk 3 above)
Comment (Pnom) [reply src/mtx/utils/deviceType.ts]: fixed

**Review by DarkPark — APPROVED** [review]

File: src/mtx/utils/deviceType.ts [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    } else {
+        // 0000.00.000-1111
+        type.push(nibbles9[nibbles[10]] ?? DEVICE_TYPE_INVALID_CHAR);
+        type.push(nibbles9[nibbles[11]] ?? DEVICE_TYPE_INVALID_CHAR);
+        type.push(nibbles9[nibbles[12]] ?? DEVICE_TYPE_INVALID_CHAR);
+        deviceProtocolIndex = 13;
+    }
+
+    const deviceProtocolNibble = nibbles[deviceProtocolIndex];
+    if ( deviceProtocolNibble && deviceProtocolNibble !== 0 ) {
```
Comment (DeadByElpy) [inline src/mtx/utils/deviceType.ts:109]:
```suggestion

    if ( deviceProtocolNibble && deviceProtocolNibble !== 0 ) {
```

File: src/mtx/utils/deviceType.ts [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+
+    // 4: 0000.00.110-0000
+    type.push(nibbles6[nibbles[6]] ?? DEVICE_TYPE_INVALID_CHAR);
+    type.push(nibbles7[nibbles[7]] ?? DEVICE_TYPE_INVALID_CHAR);
+
+    // 5: 0000.00.001-0000
+    const revision = nibbles[8];
+    type.push(nibbles8[nibbles[9]] ?? DEVICE_TYPE_INVALID_CHAR);
+
+    type.push('-');
```
Comment (DeadByElpy) [inline src/mtx/utils/deviceType.ts:86]:
```suggestion

    type.push(nibbles8[nibbles[9]] ?? DEVICE_TYPE_INVALID_CHAR);
    type.push('-');
```

**Pull request merged**